frontend/src/components/common/Resource/Resource.tsx

```diff
@@ -894,3 +895,175 @@ export function ConditionsSection(props: { resource: KubeObjectInterface | null
     </SectionBox>
   );
 }
+
+export interface VolumeSectionProps {
+  volume?: KubeVolume;
+  resource: KubeObjectInterface | null;
+}
+
+export interface PrintVolumeLinkProps {
+  name?: string;
+  configMap?: {
+    name: string;
+    defaultMode: number;
+  };
+  persistentVolumeClaim?: {
+    claimName: string;
+    readOnly: boolean;
+  };
+}
+
+export function VolumeSection(props: VolumeSectionProps) {
+  const { resource } = props;
+
+  const volumes = resource?.spec?.volumes;
+
+  if (!volumes) {
+    return null;
+  }
+
+  const namespace = resource?.metadata?.namespace;
+
+  const { t } = useTranslation();
```

Comment:
Hooks cannot be conditionally used. This is the case in this function because depending on the value of volumes, this hook is either used or not. (Please move the hooks to the first lines, maybe after the props expansion)